### PR TITLE
docs: improve Go binary version docs

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Improve Go binary version docs

## Context:

In discussion #10941 @rarkins said:

> I think our documentation is outdated/wrong. The go authors recommended we treat the line in go.mod as a minimum version and not exact.
> 
> We do in general support override of auto detected versions using the `constraints` object and hopefully they works out of the box already. Eg configure constraints.go=1.15

I've rewritten the text so that we use "minimum version" instead of "pinned version" as the concept.
I don't know if we want to add text about the `constraints.go=1.15` thing?

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
Maybe add a contraint sample to pin a version?

**Comment by HonkingGoose** [comment]
> Maybe add a constraint sample to pin a version?

Are we sure that Renovate supports this properly? 

I'll wait for confirmation before adding the constraint sample. 😉

**Comment by rarkins** [comment]
Confirmed that:
- if config.constraints.go is configure then it will be used,
- else latest for version will be used

I think we possibly don't even look at the version in go.mod as it's just a "minimum version"

File: docs/usage/golang.md [hunk 1]
```diff
@@ -40,11 +40,13 @@ Renovate will commit all files changed within the `vendor/` folder.
...
 ### Go binary version
 
-By default, Renovate will keep up with the very latest version of `go`.
+By default, Renovate will keep up with the latest version of the `go` binary.
 
-You can "pin" the `go` version that Renovate uses.
-Say you want Renovate to use Go version 1.14, you can do this by adding `go 1.14` to your `go.mod` file.
-We do not support pinning Go versions to a specific patch level, so you cannot use `go 1.14.12`, but you can use `go 1.14` in your `go.mod` file.
+You can set a contraint in your `go.mod` file to force Renovate to use a specific version.
+As an example, say you want Renovate to use the latest patch version of the `1.16` Go binary, then you would put `contstraints.go=1.16` in your `go.mod` file.
```
Comment (HonkingGoose) [inline docs/usage/golang.md:46]: Please check if this `contraints` example is correct.

I've bumped the version of Go in the examples to `1.16.6` or `1.16`, as that is the latest binary. 😄

Comment (rarkins) [reply docs/usage/golang.md:46]: This should be renovate config, not go.mod.

File: docs/usage/golang.md [hunk 2]
```diff
@@ -40,11 +40,19 @@ Renovate will commit all files changed within the `vendor/` folder.
...
-Say you want Renovate to use Go version 1.14, you can do this by adding `go 1.14` to your `go.mod` file.
-We do not support pinning Go versions to a specific patch level, so you cannot use `go 1.14.12`, but you can use `go 1.14` in your `go.mod` file.
+You can force Renovate to use a specific version of Go by setting a constraint.
+As an example, say you want Renovate to use the latest patch version of the `1.16` Go binary, you'd put this in your Renovate config:
+
+```json
+  "constraints": {
+    "go": "1.16"
+  }
+```
```
Comment (HonkingGoose) [inline docs/usage/golang.md:52]: @rarkins Is this what you meant?

Our docs say that `constraints` is only supported for Python:

> For now this only supports `python`, other compatibility restrictions will be added in the future.

https://docs.renovatebot.com/configuration-options/#constraints

Comment (rarkins) [reply docs/usage/golang.md:52]: That needs updating too! Here's what I find:

image

Comment (rarkins) [reply docs/usage/golang.md:52]: Fixed

Comment (HonkingGoose) [reply docs/usage/golang.md:52]: @rarkins Thank you for fixing the other references to the `constraints`! 😄

**Pull request merged**